db/seeds.d/16-mail_notifications.rb

```diff
@@ -60,8 +59,14 @@
     mail.attributes = notification
     mail.save! if mail.changed?
   else
-    created_notification = MailNotification.create(notification)
+    created_notification = notification[:name] == 'puppet_error_state' ? PuppetError.create(notification) : MailNotification.create(notification)
     raise ::Foreman::Exception.new(N_("Unable to create mail notification: %s"),
                                    format_errors(created_notification)) if created_notification.nil? || created_notification.errors.any?
   end
 end
+
+unless PuppetError.find_by_name('puppet_error_state').present?
```

Comment:
The`puppet_error_state` notification in the database need to become `PuppetError` subclass.
I tried to use `becomes!` with no success.